### PR TITLE
Add Release Notes collect changelog script Java

### PR DESCRIPTION
**Pull request by samvaity**
This PR adds a script to collect changelog references and details for all the packages releasing in the given month to help when preparing the monthly release notes.

- This script currently outputs all the entries for detailed changelogs. Should the release notes be a copy of the changelogs?
  Consider extracting only `New Features` and `Key Bug Fixes`?  
 _Currently, it will need a manual intervention to parse out important content out of the detailed per package changelog notes._
- Java needs a special grouping for Spring packages to be Azure Spring Cloud or Azure Spring Boot. [Understand the grouping from Spring team]
_Spring team suggests keeping spring packages aggregation out of this and they will prepare the PR for release notes themselves._

Closes: #15374

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

File: eng/scripts/Collect-Changelogs.ps1 [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+                $ReleaseNotes += $_.Replace("###", "####")
+                $ReleaseNotes += "`n"
+            }
+            $ReleaseNotes += "`n"
+        }
+    }
+}
+
+return $InstallNotes, $ReleaseNotes
```
Comment (weshaggard) [inline eng/scripts/Collect-Changelogs.ps1:48]: Do you plan to call this manually and filter it down to just the set of packages you are interested in?

Comment (samvaity) [reply eng/scripts/Collect-Changelogs.ps1:48]: Yes, currently that is the plan.

**Pull request merged**